src/wasm-builder.h

```diff
@@ -1262,7 +1266,7 @@ class Builder {
     if (curr->type.isTuple() && curr->type.isDefaultable()) {
       return makeConstantExpression(Literal::makeZeros(curr->type));
     }
-    if (curr->type.isNullable()) {
+    if (curr->type.isNullable() && curr->type.isNull()) {
```

Comment:
Why can't we just have the 2nd condition? (How can this have a null type, but that type is not nullable?)

Reply:
For example this could be a `Call` of a function that returns `(ref none)`. At runtime that call will never finish, but it is still a valid instruction with type `(ref none)`. In that case we couldn't replace it with a `ref.null` because that would require changing the type to `(ref null none)`.

Reply:
Interesting...

So `(ref none)` is kind of like an unreachable type in wasm, that can actually be used as a function's return type..? Perhaps we could use that.

Reply:
Yeah exactly. Although there are three of them (`(ref none)`, `(ref nofunc)`, and `(ref noextern)`), they don't actually put validation into unreachable mode, and the rest of the code is already set up to handle `Type::unreachable` correctly, so I think the best way to use them would be to translate them to `unreachable` or place an `(unreachable)` after any instruction that produces one.